Add recipe for difflib.

dieggsy opened the pull request:
### Brief summary of what the package does

Provides helpers for computing deltas between sequences. It's a port of [python's difflib](https://github.com/python/cpython/blob/master/Lib/difflib.py) to emacs-lisp that, in the words of the original library, provides "human friendly" diff tools.

### Direct link to the package repository

https://github.com/dieggsy/difflib.el

### Your association with the package

Maintainer

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

Please confirm with `x`:

- [X] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses). 
- [X] I've read [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)
- [X] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [X] My elisp byte-compiles cleanly
- [X] `M-x checkdoc` is happy with my docstrings **(mostly)**
- [X] I've built and installed the package using the instructions in [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)

purcell commented:
Thanks - great idea! Quick feedback:

- `difflib-pythonic-strings` is a bad name for the corresponding variable.
- Does it make sense to autoload [difflib-sequence-matcher](https://github.com/dieggsy/difflib.el/blob/d374dbd2a03b44a9852989b72ea585bc0b1f91e6/difflib.el#L50)? And does that work? i.e. does it allow people to write sequence matchers elsewhere such that `difflib` is not actually loaded until one of its methods is called? If not, you should probably just have users `require` this library and do away with the autoload.
- I'd simply write the body of [this function](https://github.com/dieggsy/difflib.el/blob/d374dbd2a03b44a9852989b72ea585bc0b1f91e6/difflib.el#L706-L707) like `(string-match-p (pat "\s*$") line)`.

dieggsy commented:
- Changed to `difflib-handle-chars-as-strings`, which is more consistent with the docstring.
- It does work, (the object constructor and a couple other things like type checker get autoloaded) and I think it makes sense since you'd only ever used one of `difflib-sequence-matcher`'s methods on an instance of `difflib-sequence-matcher` itself, so making the sequence-matcher class accessible through an autoload would let people play with the class/methods in their init if they want without needing a `require`.
- Good catch, changed `difflib-is-character-junk-p` and `difflib-is-line-junk-p` to follow the simpler syntax, assuming you meant "or".

purcell commented:
Excellent. And yes, I forgot the `or`, sorry! Merging now. :-)
